SpatialGDK/Source/SpatialGDK/Private/Utils/Interest/NetCullDistanceInterest.cpp

```diff
@@ -0,0 +1,335 @@
+
+		QueryConstraint RadiusConstraint;
+		RadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ FullFrequencyCheckoutRadius };
+
+		QueryConstraint CheckoutRadiusConstraint;
+		CheckoutRadiusConstraint.AndConstraint.Add(RadiusConstraint);
+		CheckoutRadiusConstraint.AndConstraint.Add(ComponentConstraint);
+
+		AddToFrequencyConstraintMap(FullFrequencyHz, CheckoutRadiusConstraint, FrequencyToConstraints);
+
+		// Add interest query for specified distance/frequency pairs
+		for (const auto& DistanceFrequencyPair : SpatialGDKSettings->InterestRangeFrequencyPairs)
+		{
+			float CheckoutRadius = MaxCheckoutRadiusMeters * DistanceFrequencyPair.DistanceRatio;
+
+			QueryConstraint FrequencyRadiusConstraint;
+			FrequencyRadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ CheckoutRadius };
+
+			QueryConstraint FrequencyCheckoutRadiusConstraint;
+			FrequencyCheckoutRadiusConstraint.AndConstraint.Add(FrequencyRadiusConstraint);
+			FrequencyCheckoutRadiusConstraint.AndConstraint.Add(ComponentConstraint);
+
+			AddToFrequencyConstraintMap(DistanceFrequencyPair.Frequency, FrequencyCheckoutRadiusConstraint, FrequencyToConstraints);
+		}
+	}
+
+	FrequencyConstraints CheckoutConstraints;
+
+	// De dupe across frequencies.
+	for (auto& FrequencyConstraintsPair : FrequencyToConstraints)
+	{
+		TSchemaOption<float> SpatialFrequency = FrequencyConstraintsPair.Key == FullFrequencyHz ? FullFrequencyOptional : TSchemaOption<float>(FrequencyConstraintsPair.Key);
+		if (FrequencyConstraintsPair.Value.Num() == 1)
+		{
+			CheckoutConstraints.Add({ SpatialFrequency, FrequencyConstraintsPair.Value[0] });
+			continue;
+		}
+		QueryConstraint RadiusDisjunct;
+		RadiusDisjunct.OrConstraint.Append(FrequencyConstraintsPair.Value);
+		CheckoutConstraints.Add({ SpatialFrequency, RadiusDisjunct });
+	}
+
+	return CheckoutConstraints;
+}
+
+void NetCullDistanceInterest::AddToFrequencyConstraintMap(const float Frequency, const QueryConstraint& Constraint, FrequencyToConstraintsMap& OutFrequencyToConstraints)
+{
+	// If there is already a query defined with this frequency, group them to avoid making too many queries down the line.
+	// This avoids any extra cost due to duplicate result types across the network if they are large.
+	TArray<QueryConstraint>& ConstraintList = OutFrequencyToConstraints.FindOrAdd(Frequency);
+	ConstraintList.Add(Constraint);
+}
+
+QueryConstraint NetCullDistanceInterest::GetDefaultCheckoutRadiusConstraint()
+{
+	const float MaxDistanceSquared = GetDefault<USpatialGDKSettings>()->MaxNetCullDistanceSquared;
+
+	// Use AActor's ClientInterestDistance for the default radius (all actors in that radius will be checked out)
+	const AActor* DefaultActor = Cast<AActor>(AActor::StaticClass()->GetDefaultObject());
+
+	float DefaultDistanceSquared = DefaultActor->NetCullDistanceSquared;
+
+	if (MaxDistanceSquared > FLT_EPSILON && DefaultDistanceSquared > MaxDistanceSquared)
```

Comment:
MaxDistanceSquared isn't calculated, so using epsilon isn't necessary.

Reply:
But doesn't really matter

Reply:
danny's counter example was if someone put in negative 0 which was convincing enough for me

Reply:
That person would be fired